src/VisualStudio/CSharp/Impl/Interactive/OpenInteractiveWindowCommand.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.VisualStudio.Extensibility;
+using Microsoft.VisualStudio.Extensibility.Commands;
+using Microsoft.VisualStudio.Extensibility.VSSdkCompatibility;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.CSharp.Interactive;
+
+/// <summary>
+/// Implements View/Other Windows/C# Interactive command.
+/// </summary>
+[VisualStudioContribution]
+internal class OpenInteractiveWindowCommand(
+    MefInjection<IThreadingContext> threadingContext,
+    MefInjection<CSharpVsInteractiveWindowProvider> interactiveWindowProvider) : Command
+{
+    private IThreadingContext? _threadingContext;
+    private CSharpVsInteractiveWindowProvider? _interactiveWindowProvider;
+
+    public override CommandConfiguration CommandConfiguration => new("%CSharpLanguageServiceExtension.OpenInteractiveWindow.DisplayName%")
+    {
+        Placements = new[] { CommandPlacement.KnownPlacements.ViewOtherWindowsMenu.WithPriority(0x8000) },
+        // TODO: Shortcuts https://github.com/dotnet/roslyn/issues/3941
+    };
+
+    public override async Task InitializeAsync(CancellationToken cancellationToken)
+    {
+        await base.InitializeAsync(cancellationToken).ConfigureAwait(false);
+
+        _threadingContext = await threadingContext.GetServiceAsync().ConfigureAwait(false);
+        _interactiveWindowProvider = await interactiveWindowProvider.GetServiceAsync().ConfigureAwait(false);
+    }
+
+    public override async Task ExecuteCommandAsync(IClientContext context, CancellationToken cancellationToken)
+    {
+        Contract.ThrowIfNull(_threadingContext);
+        Contract.ThrowIfNull(_interactiveWindowProvider);
```

Comment:
Why not just store `MefInjection<T>` fields and access them here?

Reply:
We could do that. I was trying to avoid getting the service each time we execute the command, but perhaps that's unnecessary.

Reply:
The implementation uses `AsyncLazy<T>` which caches the result of calling `GetValueAsync()`.